pkg/reconciler/pipelinerun/pipelinerun_test.go

```diff
@@ -4759,10 +4759,112 @@ func TestReconcileWithPipelineResults(t *testing.T) {
 	for _, tc := range testCases {
 		t.Run(tc.name, func(t *testing.T) {
 			runTestReconcileWithPipelineResults(t, tc.embeddedVal)
+			runTestReconcileWithFinallyResults(t, tc.embeddedVal)
 		})
 	}
 }
 
+func runTestReconcileWithFinallyResults(t *testing.T, embeddedStatus string) {
+	names.TestingSeed()
+	ps := []*v1beta1.Pipeline{parse.MustParsePipeline(t, `
+metadata:
+  name: test-pipeline
+  namespace: foo
+spec:
+  results:
+  - description: pipeline result
+    name: result
+    value: $(finally.a-task.results.aResult)
+  tasks:
+  - name: b-task
+    params:
+    - name: bParam
+      value: $(tasks.a-task.results.aResult)
+    taskRef:
+      name: b-task
+  finally:
+  - name: a-task
+    taskRef:
+      name: a-task
+`)}
+	trs := []*v1beta1.TaskRun{mustParseTaskRunWithObjectMeta(t,
+		taskRunObjectMeta("test-pipeline-run-different-service-accs-a-task", "foo",
+			"test-pipeline-run-different-service-accs", "test-pipeline", "a-task", true),
+		`
+spec:
+  resources: {}
+  serviceAccountName: test-sa
+  taskRef:
+    name: hello-world
+  timeout: 1h0m0s
+status:
+  conditions:
+  - status: "True"
+    type: Succeeded
+  taskResults:
+  - name: aResult
+    value: aResultValue
+`)}
+	prs := []*v1beta1.PipelineRun{parse.MustParsePipelineRun(t, `
+metadata:
+  name: test-pipeline-run-different-service-accs
+  namespace: foo
+spec:
+  pipelineRef:
+    name: test-pipeline
+  serviceAccountName: test-sa-0
+status:
+  completionTime: "2022-01-01T00:00:00Z"
+  conditions:
+  - message: All Tasks have completed executing
+    reason: Succeeded
+    status: "True"
+    type: Succeeded
+  pipelineResults:
+  - name: result
+    value: aResultValue
```

Comment:
It looks like you're creating a pipelinerun that already has results populated, and testing that it has results populated after being reconciled. I think instead we want to create a finally taskrun that has results populated, and then test that after the pipelinerun is reconciled, it has pipeline results populated. does that make sense?

Reply:
need to make sure tekton is implenting syntax, this that has been highlighted is input, not output

Reply:
delete `pipelineResults`

Reply:
done